src/communication/NetworkClient.ts

```diff
@@ -69,21 +68,37 @@ const throwApiError = run(
   },
   findProperty =>
     function throwApiError(cause: unknown) {
-      if (findProperty(cause, 'response') && cause.response != undefined) {
-        throw ApiError.createFromResponse(cause.response as AxiosResponse<any>);
-      }
       throw new ApiError(findProperty(cause, 'message') ? String(cause.message) : 'An unknown error has occurred');
     },
 );
 
+/**
+ * Checks whether an API error needs to be thrown based on the passed result.
+ */
+async function processFetchResponse(response: ResponseWithIdempotencyKey) {
+  // Request was successful, but no content was returned.
+  if (response.status == 204) return true;
+  // Request was successful and content was returned.
+  const body = await response.json();
+  if (Math.floor(response.status / 100) == 2) {
+    return body;
+  }
+  // Request was not successful, but the response body contains an error message.
+  if (body) {
```

Comment:
Can we replace this truth-y check with something else? Are we checking for `undefined` here? Or null-ish?